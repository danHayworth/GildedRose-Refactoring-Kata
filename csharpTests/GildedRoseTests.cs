using ApprovalTests;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using NUnit.Framework;
using System.Collections.Generic;

namespace csharp.Tests
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    [TestClass()]
    public class GildedRoseTests
    {
        [TestMethod()]
        public void GildedRoseTest()
        {
            IList<Item> items = new List<Item>(Program.Items);
            GildedRose app = new GildedRose(items);
            app.UpdateQuality();
            Approvals.VerifyAll<Item>(items, label: "");           
        }

        [TestMethod()]
        public void UpdateQualityTest()
        {
            CombinationApprovals.VerifyAllCombinations(
                getStringItem,
                new string[] { "Aged Brie", "Backstage passes to a TAFKAL80ETC concert", "Sulfuras, Hand of Ragnaros", "+5 Dexterity Vest", "Elixir of the Mongoose" },
                new int[] {-1, 0, 11 },
                new int[] {0, 1, 49, 50 }
                );
        }
        private string getStringItem(string name, int sellIn, int quality)
        {           
            IList<Item> newItems = new List<Item>() { new Item { Name = name, SellIn = sellIn, Quality = quality } };
            GildedRose gilded = new GildedRose(newItems);
            gilded.UpdateQuality();
            return newItems[0].ToString();
            
        }
    }
}