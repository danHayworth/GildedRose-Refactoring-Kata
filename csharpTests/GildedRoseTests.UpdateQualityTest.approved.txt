[Aged Brie,-1,0] => Aged Brie, -2, 2
[Aged Brie,-1,1] => Aged Brie, -2, 3
[Aged Brie,-1,49] => Aged Brie, -2, 50
[Aged Brie,-1,50] => Aged Brie, -2, 50
[Aged Brie,0,0] => Aged Brie, -1, 2
[Aged Brie,0,1] => Aged Brie, -1, 3
[Aged Brie,0,49] => Aged Brie, -1, 50
[Aged Brie,0,50] => Aged Brie, -1, 50
[Aged Brie,11,0] => Aged Brie, 10, 1
[Aged Brie,11,1] => Aged Brie, 10, 2
[Aged Brie,11,49] => Aged Brie, 10, 50
[Aged Brie,11,50] => Aged Brie, 10, 50
[Backstage passes to a TAFKAL80ETC concert,-1,0] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,1] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,49] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,-1,50] => Backstage passes to a TAFKAL80ETC concert, -2, 0
[Backstage passes to a TAFKAL80ETC concert,0,0] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,1] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,49] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,0,50] => Backstage passes to a TAFKAL80ETC concert, -1, 0
[Backstage passes to a TAFKAL80ETC concert,11,0] => Backstage passes to a TAFKAL80ETC concert, 10, 1
[Backstage passes to a TAFKAL80ETC concert,11,1] => Backstage passes to a TAFKAL80ETC concert, 10, 2
[Backstage passes to a TAFKAL80ETC concert,11,49] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Backstage passes to a TAFKAL80ETC concert,11,50] => Backstage passes to a TAFKAL80ETC concert, 10, 50
[Sulfuras, Hand of Ragnaros,-1,0] => Sulfuras, Hand of Ragnaros, -1, 0
[Sulfuras, Hand of Ragnaros,-1,1] => Sulfuras, Hand of Ragnaros, -1, 1
[Sulfuras, Hand of Ragnaros,-1,49] => Sulfuras, Hand of Ragnaros, -1, 49
[Sulfuras, Hand of Ragnaros,-1,50] => Sulfuras, Hand of Ragnaros, -1, 50
[Sulfuras, Hand of Ragnaros,0,0] => Sulfuras, Hand of Ragnaros, 0, 0
[Sulfuras, Hand of Ragnaros,0,1] => Sulfuras, Hand of Ragnaros, 0, 1
[Sulfuras, Hand of Ragnaros,0,49] => Sulfuras, Hand of Ragnaros, 0, 49
[Sulfuras, Hand of Ragnaros,0,50] => Sulfuras, Hand of Ragnaros, 0, 50
[Sulfuras, Hand of Ragnaros,11,0] => Sulfuras, Hand of Ragnaros, 11, 0
[Sulfuras, Hand of Ragnaros,11,1] => Sulfuras, Hand of Ragnaros, 11, 1
[Sulfuras, Hand of Ragnaros,11,49] => Sulfuras, Hand of Ragnaros, 11, 49
[Sulfuras, Hand of Ragnaros,11,50] => Sulfuras, Hand of Ragnaros, 11, 50
[+5 Dexterity Vest,-1,0] => +5 Dexterity Vest, -2, 0
[+5 Dexterity Vest,-1,1] => +5 Dexterity Vest, -2, 0
[+5 Dexterity Vest,-1,49] => +5 Dexterity Vest, -2, 47
[+5 Dexterity Vest,-1,50] => +5 Dexterity Vest, -2, 48
[+5 Dexterity Vest,0,0] => +5 Dexterity Vest, -1, 0
[+5 Dexterity Vest,0,1] => +5 Dexterity Vest, -1, 0
[+5 Dexterity Vest,0,49] => +5 Dexterity Vest, -1, 47
[+5 Dexterity Vest,0,50] => +5 Dexterity Vest, -1, 48
[+5 Dexterity Vest,11,0] => +5 Dexterity Vest, 10, 0
[+5 Dexterity Vest,11,1] => +5 Dexterity Vest, 10, 0
[+5 Dexterity Vest,11,49] => +5 Dexterity Vest, 10, 48
[+5 Dexterity Vest,11,50] => +5 Dexterity Vest, 10, 49
[Elixir of the Mongoose,-1,0] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose,-1,1] => Elixir of the Mongoose, -2, 0
[Elixir of the Mongoose,-1,49] => Elixir of the Mongoose, -2, 47
[Elixir of the Mongoose,-1,50] => Elixir of the Mongoose, -2, 48
[Elixir of the Mongoose,0,0] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose,0,1] => Elixir of the Mongoose, -1, 0
[Elixir of the Mongoose,0,49] => Elixir of the Mongoose, -1, 47
[Elixir of the Mongoose,0,50] => Elixir of the Mongoose, -1, 48
[Elixir of the Mongoose,11,0] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose,11,1] => Elixir of the Mongoose, 10, 0
[Elixir of the Mongoose,11,49] => Elixir of the Mongoose, 10, 48
[Elixir of the Mongoose,11,50] => Elixir of the Mongoose, 10, 49
